tests/test_startlette_request_form.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+from fastapi import FastAPI, UploadFile
+from fastapi.testclient import TestClient
+
+app = FastAPI()
+
+
+@app.post("/route-with-default-limit")
+def files_low(files: List[UploadFile]):
+    return files  # pragma: nocover
+
+
+@app.post("/route-with-custom-limit", max_files=2000)
+def files_high(files: List[UploadFile]):
+    return files  # pragma: nocover
+
+
+client = TestClient(app)
+
+
+def test_max_files_setting_default():
+    # Test Starlette default remains in place
+    files = [("files", ("foo.txt", "foo"))] * 1500
+    response = client.post("/route-with-default-limit", files=files)
+    # Fails with {'detail': 'Too many files. Maximum number of files is 1000.'}
+    assert response.status_code == 400
```

Comment:
Why don't you assert against error details ? Like this is too broad.

Reply:
Sound sensible will include! It is annoying to couple against the text in starlette